src/main/java/com/eatthepath/uuid/FastUUID.java

```diff
@@ -107,6 +109,35 @@ private FastUUID() {
      * described in {@link UUID#toString()}
      */
     public static UUID parseUUID(final CharSequence uuidSequence) {
+        if (USE_JDK_UUID_FROM_STRING && uuidSequence instanceof String) {
+            // OpenJDK 15 and newer use a faster method for parsing UUIDs
+            return UUID.fromString((String) uuidSequence);
+        }
+
+        return parseUUIDInternal(uuidSequence);
+    }
+
+    /**
+     * Parses a UUID from the given string. The string must represent a UUID as described in
+     * {@link UUID#toString()}.
+     *
+     * @param uuidString the string from which to parse a UUID
+     *
+     * @return the UUID represented by the given string
+     *
+     * @throws IllegalArgumentException if the given string does not conform to the string representation as
+     * described in {@link UUID#toString()}
+     */
+    public static UUID parseUUID(final String uuidString) {
```

Comment:
Because JDK's method only works with `String`.

So we can:

1. Change the signature of our method to `parseUUID(String)` and possibly break compatibility;
2. Call `CharSequence#toString()` but that will lead to memory copies in any non-`String` objects that implement `CharSequence`;
3. Add a method that uses `String` (so it can be passed to JDK's in case we're running on >= 15) and keep the `CharSequence` there too and shared the common code path between the two;
4. Number 3 + we check if the `CharSequence` is a `String` so we can pass it to JDK's, but we keep using our implementation in case it's not (since it performs closely anyway).

`instanceof` has basically no cost on modern JVMs anyway, so I chose to go with 4.

Reply:
Ah—I think I see where you're going here. If I can reword it to make sure we're on the same page, I think you're saying that you want to add a `String`-specific variant because that will lower the cost of passing through to `UUID#fromString` under Java 15 and newer.

If I'm understanding that right, I'd gently push back that I think it's okay to continue using fast-uuid's parser even under Java 15. My rationale is that if somebody is working with non-`String` `CharSequences`, they probably have some specific thing they're doing that means they want to avoid `String` conversion in the first place, and using the sliiiiightly slower `FastUuid#parseUUID(CharSequence)` will still be faster for those users than converting to a `String`, then using the slightly faster `UUID#fromString(String)`.

As you say, using `instanceof` in `FastUuid#parseUUID(CharSequence)` is a negligible cost, so _most_ users would still get the benefits of using the built-in parser under Java 15 and newer, and it's only the oddballs (like me!) who have a `CharSequence`-specific use case that would be staying in the fast-uuid space.

Does that make sense? Am I missing your point?

Reply:
It's more of a compile-time optimization and maybe being able to deal with JVMs where `instanceof` isn't super-duper cheap.

If you upgrade from existing code to this, your compiler will link your calls that use String-typed references to the new method. So

```java
String uuidString = someMethodThatReturnsString();
// this will now link directly to the String method instead of the CharSequence one without the user having to change any code
UUID id = FastUUID.parseUUID(uuidString);

CharSequence uuidChars = someMethodThatReturnsCharSequence();
// this will still link to the CharSequence method
UUID idFromCharSequence = FastUUID.parseUUID(uuidChars);
``` 

If `uuidChars` happens to actually be a String, we can still leverage the speed boost by checking that and calling JDK's `UUID#fromString`. 

Providing the overload with String allows us to skip that step without the user changing any code because upon compilation, they'd link to the new method if the reference is String-typed. 

A CharSequence reference to a CharSequence non-String object will still just use the existing parser just fine. There is no conversion involved. 

I see this as a win-win change. The only downside is that `FastUUID.parseUUID(null)` would not compile anymore. You'd have to cast the null to either CharSequence or String.

For someone using fast-uuid indirectly, the type-check to String inside the CharSequence overload still allows for the performance boost in Java 15 even without recompilation.